instrument/build.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package instrument
+
+import (
+	"errors"
+	"log"
+	"runtime"
+	"sync"
+	"time"
+)
+
+var (
+	// Revision is the VCS revision associated with this build. Overridden using ldflags
+	// at compile time. Example:
+	// $ go build -ldflags "-X github.com/m3db/m3x/instrument.Revision=abcdef" ...
+	// Adapted from: https://www.atatus.com/blog/golang-auto-build-versioning/
+	Revision = "unknown"
+
+	// Branch is the VCS branch associated with this build.
+	Branch = "unknown"
+
+	// BuildDate is the date this build was created.
+	BuildDate = "unknown"
+
+	// LogBuildInfoAtStartup controls whether we log build information at startup.
+	LogBuildInfoAtStartup = false
+
+	// goVersion is the current runtime version.
+	goVersion = runtime.Version()
+
+	// metricName is the emitted metric's name.
+	metricName = "build-information"
+)
+
+var (
+	errAlreadyStarted = errors.New("reporter already started")
+	errNotStarted     = errors.New("reporter not started")
+)
+
+// LogBuildInformation logs the build information to the provided logger.
+func LogBuildInformation() {
```

Comment:
super duper nit: maybe `LogBuildInfo` instead of `LogBuildInformation` to match `LogBuildInfoAtStartup`